packages/lesmis-server/src/routes/VerifyEmailRoute.ts

```diff
@@ -0,0 +1,29 @@
+/*
+ * Tupaia
+ *  Copyright (c) 2017 - 2021 Beyond Essential Systems Pty Ltd
+ */
+
+import { Request, Response, NextFunction } from 'express';
+import { Route } from '@tupaia/server-boilerplate';
+import { MeditrakConnection } from '../connections';
+
+export class VerifyEmailRoute extends Route {
+  private readonly meditrakConnection: MeditrakConnection;
+
+  constructor(req: Request, res: Response, next: NextFunction) {
+    super(req, res, next);
+
+    this.meditrakConnection = new MeditrakConnection(req.session);
+  }
+
+  async buildResponse() {
+    const { emailToken } = this.req.params;
+
+    const response = await this.meditrakConnection.verifyEmail(emailToken);
+    if (response.emailVerified === 'true') {
+      return response;
+    }
+
+    throw Error('Email was not correctly verified');
```

Comment:
I wonder if the error message meditrak-server returns might be more informative?

Reply:
Have checked and it's also very generic (`'Email address could not be verified'`), but might as well try to use it in case we change that logic centrally later. I'm also fairly sure that it will throw an error on line 22 if the verification doesn't work properly, rather than having a response that doesn't have `emailVerified` - could you test to make sure the failing case works? E.g. pass in an invalid token

Reply:
I updated it to just return the response which definitely makes more sense. I tested the failing case and works well.